packages/dev/core/src/Shaders/screenSpaceReflection.fragment.fx

```diff
@@ -174,3 +196,355 @@ void main()
+        info.coords = uv;
+    }
+
+    return info;
+}
+
+void main(void)
+{
+    #ifdef SSR_SUPPORTED
+
+    // *************** Get data from samplers ***************
+
+    vec4 original = texture2D(textureSampler, vUV);
+    vec3 spec = toLinearSpace(texture2D(specularSampler, vUV).rgb);
+
+    if (dot(spec, vec3(1.0)) <= 0.0){
+        gl_FragColor = texture2D(textureSampler, vUV); // no reflectivity, no need to compute reflection
+        return;
+    }
+
+    float roughness = 1.0 - texture2D(specularSampler, vUV).a;
+
+    // Get coordinates of the direction of the reflected ray
+    // according to the pixel's position and normal.
+    vec3 unitNormal = normalize((texture2D(normalSampler, vUV)).xyz);
+    vec3 position = (view * texture2D(positionSampler, vUV)).xyz;
+
+    vec3 unitPosition = normalize(position);
+    vec3 reflected = normalize(reflect(unitPosition, unitNormal)); // incident direction = unit position in camera space
+
+    // *************** Compute reflection info  ***************
+    ReflectionInfo info;
+
+    // hash(position) represents a random vector3, jitt represents a bias to simulate roughness (light deviation)
+    vec3 jitt = mix(vec3(0.0), hash(texture2D(positionSampler, vUV).xyz), roughness) * roughnessFactor;
+
+    #ifdef RIGHT_HANDED_SCENE
+        if (position.z < -distanceFade || distanceFade == 0.0){ // no need to compute reflection, the point we are evaluating is further than the distanceFade
+            #if defined(BACKUP_TEXTURE_SKYBOX) || defined(BACKUP_TEXTURE_PROBE)
+                info.coords = vUV;
+                info.visibility = 0.0;
+                info.miss = true;
+                info.visibilityBackup = 1.0;
+            #else
+                gl_FragColor = texture2D(textureSampler, vUV);
+                return;
+            #endif
+        } else {
+            vec2 texSize = gl_FragCoord.xy / vUV;
+            info = getReflectionInfo2DRayMarching(reflected + jitt, position, texSize);
+        }
+    #else // if left handed scene
+        if (position.z > distanceFade || distanceFade == 0.0){ // no need to compute reflection, the point we are evaluating is further than the distanceFade
+            #if defined(BACKUP_TEXTURE_SKYBOX) || defined(BACKUP_TEXTURE_PROBE)
+                info.coords = vUV;
+                info.visibility = 0.0;
+                info.miss = true;
+                info.visibilityBackup = 1.0;
+            #else
+                gl_FragColor = texture2D(textureSampler, vUV); 
+                return;
+            #endif
+        } else {
+            vec2 texSize = gl_FragCoord.xy / vUV;
+            info = getReflectionInfo2DRayMarching(reflected + jitt, position, texSize);
+
+        }
+    #endif
+
+    // *************** Get reflection color ***************
+
+    vec3 reflectedColor;
+    
+    #if defined(BACKUP_TEXTURE_SKYBOX) || defined(BACKUP_TEXTURE_PROBE)
+        if (backupOnlyWhenTooSpecular && dot(spec, vec3(1.0))/3.0 > 0.9) {

```

Comment:
Do we really need this?

That's a new constant that the user will probably want to tweak, so more parameters...

Also, I think the parameter is not very clear (for me at least), something like `forceBackupOnlyWhenTooSpecular` would be clearer.

Reply:
It may be needed when the user defines mirror like materials. SSR and probe don't match well, and it's prettier to use only one method rather than a mix of methods in this case.

What do you think about letting the user choosing the value, instead of using a boolean ?

Reply:
Yes, let's allow the user to change the 0.9 value instead of having `backupOnlyWhenTooSpecular`.